pkg/apis/controlplane/v1beta1/conversion.go

```diff
@@ -124,13 +126,20 @@ func Convert_v1beta1_GroupMemberPod_To_controlplane_GroupMember(in *GroupMemberP
 	return nil
 }
 
-func Convert_controlplane_GroupMember_To_v1beta1_GroupMemberPod(in *controlplane.GroupMember, out *GroupMemberPod, s conversion.Scope) error {
+// Convert_controlplane_GroupMember_To_v1beta1_GroupMemberPod converts controlplane GroupMember to v1beta1 GroupMember
+// based on whether it's required to include Pod reference in the result. We must not include Pod reference when the
+// conversion is called for an AddressGroup as agents don't expect it in v1beta1 version.
+// This function doesn't match the pattern of conversion function which requires the last parameter to be
+// conversion.Scope so won't be registered to schema.
```

Comment:
BTW, is there any advantage to having conversion functions for non resources registered to the schema? I understand why `Convert_controlplane_AddressGroup_To_v1beta1_AddressGroup` needs to be registered, but it seems conversion functions for non resources will only be called indirectly by other conversion functions. Is it for auto-generated conversion functions?

Reply:
Good question, I wondered this as well, haven't found answer so blindly followed what K8s code does.
I can take a look what it would affect later and see if we could remove them. Created #1616 to track it.